newIDE/app/src/MainFrame/index.js

```diff
@@ -303,7 +310,20 @@ class MainFrame extends React.Component<Props, State> {
   };
 
   openFromPathOrURL = (url: string, cb: Function) => {
-    const { i18n } = this.props;
+    const { i18n, shouldOpenAutosave } = this.props;
+
+    const projectFilePath = url;
+    const autoSavePath = url + '.autosave';
+    if (shouldOpenAutosave(url, autoSavePath, true)) {
```

Comment:
You specified to Flow that this function is optional, so you should verify for its existence before using it:
`if (shouldOpenAutosave || shouldOpenAutosave *url, autoSavePath, true)) {`

Otherwise, the webapp will crash.
I recommend that you try the webapp (localhost:3000 on your web-browser). You should be able to quickly check these kind of things and fix Flow errors that are complaining about it :)

Reply:
projectFilePath and autoSavePath are clear now 👍

Reply:
ah sorry I missed this. Completely forgot about the web app

Reply:
should be if `(shouldOpenAutosave && shouldOpenAutosave(url, autoSavePath, true)){`
I will also fix this on `onAutoSaveProject `